src/ring/logger.clj

```diff
@@ -6,37 +6,66 @@
     [ring.logger.messages :as messages]
     [ring.logger.protocols :refer [add-extra-middleware debug]]))
 
+(defn- assoc-end-time
+  [request timing]
+  (if timing
+    (assoc request :logger-end-time (System/currentTimeMillis))
+    request))
+
 (defn- wrap-with-logger*
   [handler {:keys [timing exceptions] :as options}]
 ;; Long ago, originally based on
 ;; https://gist.github.com/kognate/noir.incubator/blob/master/src/noir.incubator/middleware.clj
-  (fn [request]
+  (fn
+    ([request]
     (try
       (messages/starting options request)
       (messages/request-details options request)
       (messages/request-params options request)
 
       (let [response (handler request)
-            request (if timing
-                      (assoc request :logger-end-time (System/currentTimeMillis))
-                      request)]
+            request  (assoc-end-time request timing)]
         (messages/sending-response options response)
         (messages/finished options request response)
 
         response)
 
       (catch Throwable t
         (when exceptions
-          (let [request (if timing
-                          (assoc request :logger-end-time (System/currentTimeMillis))
-                          request)]
+          (let [request (assoc-end-time request timing)]
             (messages/exception options request t)))
-        (throw t)))))
+        (throw t))))
+    ([request respond raise]
+     (try
+      (messages/starting options request)
+      (messages/request-details options request)
+      (messages/request-params options request)
+
+      (handler request
+               #(let [request (assoc-end-time request timing)]
+                  (messages/sending-response options %)
+                  (messages/finished options request %)
+                  (respond %))
+               #(let [request (assoc-end-time request timing)]
+                  (messages/exception options request %)
```

Comment:
We want to log the exception by calling `messages/exception` only when `exceptions` is truthy (it's an option). It should be similar I think to what's being done in the `catch` few lines below